JoinTable.lua

```diff
@@ -8,11 +8,18 @@ function JoinTable:__init(dimension, nInputDims)
    self.nInputDims = nInputDims
 end
 
-function JoinTable:updateOutput(input)
+function JoinTable:_getPositiveDimension(input)
    local dimension = self.dimension
-   if self.nInputDims and input[1]:dim()==(self.nInputDims+1) then
-       dimension = dimension + 1
+   if dimension < 0 then
+      dimension = input:dim() + dimension + 1
```

Comment:
now that I think about it, -1 should be the last dimension, as that's the only way we can choose the last dimension via negative indexing and the rest of the dims will follow.